lib/mongodb.js

```diff
@@ -180,6 +188,21 @@ MongoDB.prototype.fromDatabase = function (model, data) {
         // Convert the Binary into String
         data[p] = data[p].toString();
       }
+    }  // check if prop contains EmbedsMany
+    else if(prop && prop.type.constructor === Array &&
```

Comment:
the problem is that the property name is not the relation's name. 
but in the previous commit ( c432359cbc71cb6db9b6f24049660adc73f07a11 ) I proposed the following:

```javascript
MongoDB.prototype.fromDatabase = function (model, data) {
  var self = this;
  var props;
  var settings;
  var objModel;
  ;
  if (!data) {
    return null;
  }
  if(typeof model === 'string'){
    model = this._models[model];
    props = model.properties;
    settings = model.settings;
  }else{
    props = model.properties;
    settings = model.settings;
  }
  for (var p in props) {
    var prop = props[p];
    if (prop && prop.type === Buffer) {
      if(data[p] instanceof mongodb.Binary) {
        // Convert the Binary into Buffer
        data[p] = data[p].read(0, data[p].length());
      }
    } else if(prop && prop.type === String) {
      if(data[p] instanceof mongodb.Binary) {
        // Convert the Binary into String
        data[p] = data[p].toString();
      }
    } // check if prop contains EmbedsMany
    else if(prop && prop.type.constructor === Array &&
        prop.type[0].toString() === model.model.toString()){
      for(var k in settings.relations){
        if(settings.relations[k].property === p){
          for(var i = 0; i < data[p].length; i++){
            data[p][i] = self.fromDatabase({
              model: prop.type[0],
              properties: prop.type[0].definition.properties,
              settings: prop.type[0].definition.settings
            },data[p][i]);
          }
        }
      }
    } // check if prop contains EmbedsOne
    else if(prop && prop.type.toString() === model.model.toString()){
      for(var k in settings.relations){
        if(settings.relations[k].property === p){
          data[p] = self.fromDatabase({
            model: prop.type,
            properties: prop.type.definition.properties,
            settings: prop.type.definition.settings
          },data[p]);
        }
      }
    }
  }
  return data;
};
```

its works, but there are many for.